src/MongoDB.Driver.Core/Core/Configuration/ClusterSettings.cs

```diff
@@ -252,5 +310,46 @@ public IServerSelector PostServerSelector
                 schemaMap: Optional.Create(schemaMap.WithDefault(_schemaMap)),
                 scheme: scheme.WithDefault(_scheme));
         }
+
+        // internal methods
+        internal ClusterType GetInitialClusterType()
+        {
+#pragma warning disable 618
+            if (_connectionModeSwitch == ConnectionModeSwitch.UseDirectConnection)
+#pragma warning restore 618
+            {
+                if (_directConnection.GetValueOrDefault())
+                {
+                    return ClusterType.Standalone;
```

Comment:
Described here: https://github.com/mongodb/specifications/blob/e56f5eceed7729f8b9b43a4a1f76c7e5840db49f/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#initial-topologytype
Before Standalone code path has been missed